### DISCUSSION TITLE
Quick FeedsAPI question.

### DISCUSSION THREAD

**Discussion by StrangeTim** [amzn/selling-partner-api-models#3710] [General] [answered]
I am in the process of converting an AMWS program over to SP-API. The current program uses Flat File feeds and does not have any code related to feed documents. I can't find a full use case for the Flat File processing ( I am following [this guide](https://developer-docs.amazon.com/sp-api/docs/feeds-api-v2021-06-30-use-case-guide) ). The 2 types of feeds I need for my program are _POST_FLAT_FILE_PRICEANDQUANTITYONLY_UPDATE_DATA_ and _POST_FLAT_FILE_FULFILLMENT_DATA_ . 

My question is this: Using SP-API, does a flat file feed upload require doing steps 2 (construct feed) and 3 (upload feed data) or can a flat file feed just be done with step 1 (Create Feed Document) and step 4 (Create Feed) ? 

2nd question: what is the purpose of having a 'construct feed' and 'create feed'. I don't feel like steps 2 and 3 do anything for me? I _think_ that step 2 may be needed for creating new products?

**Comment by StrangeTim** [comment]
Related question: How do you even submit a feed for step 3. There used to be a submit_feed method but I don't have that method available in my FeedsApi that I built! Searching this forum for submit_feed only returns 1 result. Was it deprecated? Did I build my SDK wrong?
I'm primarily confused because the use case guide specifically says that, in step 3, you need to submit the feed that you constructed in step 2, but then step 4 says to use the createFeed method. So obviously submitting a feed is not the same as 'createFeed' and the documentation generated for the createFeed method says to be sure you have submitted your feed data before calling 'createFeed' .

**Comment by MeenaAmz** [comment, marked as answer]
1. Using SP-API, does a flat file feed upload require doing steps 2 (construct feed) and 3 (upload feed data) or can a flat file feed just be done with step 1 (Create Feed Document) and step 4 (Create Feed) ? - _**For a flat file feed, step 1 is required to create the url to upload your feed, step 2 will not be complicated as with json or xsd but you will need to construct the excel file with the correct attributes, step 3 is still needed to upload this excel file to the url.**_

2. what is the purpose of having a 'construct feed' and 'create feed'. I don't feel like steps 2 and 3 do anything for me? I think that step 2 may be needed for creating new products? - **_If you look into the use case guide closely, create feed is for creating the url to upload feed document and construct feed is for constructing the feed as per specification._**
3. How do you even submit a feed for step 3. - **_Look into the [sample code](https://developer-docs.amazon.com/sp-api/docs/feeds-api-v2021-06-30-use-case-guide#sample-code-to-upload-an-xml-feed-java), this is what you would use to upload(submit) the feed document. You should change content type to match for excel. Rest of the code will be similar._**

**Reply by StrangeTim** [reply]
Thank you MeenaAmz. Follow up question to #3. The sample code builds an uploader from scratch. Does this mean that the SDK I built doesn't have a method already for uploading a constructed feed? I certainly cannot find one.
